File: LeetCode/2444/Solution.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        int n = nums.length;
        int baseIndex = 0;
        int minIndex = -1;
        int maxIndex = -1;

        long result = 0;
        for(int i=0;i<n;++i) {
            if(nums[i] < minK || nums[i] > maxK) {
                baseIndex = i+1;
                continue;
            }

            if(maxK == nums[i]) {
                maxIndex = i;
            }
            if(minK == nums[i]) {
                minIndex = i;
            }

            result += Math.max(Math.min(maxIndex, minIndex)-baseIndex+1, 0);
        }
        return result;
    }
}

File: LeetCode/2444/Solution.kt
class Solution {
    fun countSubarrays(nums: IntArray, minK: Int, maxK: Int): Long {
        var minIndex = -1
        var maxIndex = -1
        var baseIndex = 0

        var result = 0L
        for(i in 0 until nums.size) {
            if(minK > nums[i] || maxK < nums[i]) {
                baseIndex = i+1
                continue
            }
            if(maxK == nums[i]) {
                maxIndex=i
            }
            if(minK == nums[i]) {
                minIndex=i
            }
            result += max(min(maxIndex, minIndex)-baseIndex+1, 0)
        }
        return result
    }
}
